src/Symfony/Component/Security/Core/Authentication/Token/AbstractToken.php

```diff
@@ -103,7 +110,12 @@ public function __serialize(): array
      */
     public function __unserialize(array $data): void
     {
-        [$user, , , $this->attributes, $this->roleNames] = $data;
+        [$user, , , $this->attributes] = $data;
+
+        if (\array_key_exists(4, $data)) {
+            $this->roleNames = $data[4];
+        }
```

Comment:
Is it a BC layer? If so, should it be mentioned to easily find it if we search for layers to be removed in a major version for example?